src/components/NetworkSelector/useNetworkName.ts

```diff
@@ -1,22 +1,24 @@
 import useSWR from 'swr'
-import LotusRPCEngine from '@glif/filecoin-rpc-client'
+import LotusRpcEngine from '@glif/filecoin-rpc-client'
+import { useEnvironment } from '../../services/EnvironmentProvider'
 
-const fetcher = async (lotusApiAddr: string) => {
-  const lCli = new LotusRPCEngine({
-    apiAddress: lotusApiAddr
-  })
-  const network = (await lCli.request('StateNetworkName')) as string
-  if (network) return network
-
-  throw new Error('Not connected to a network')
+interface UseNetworkNameResult {
+  networkName: string
+  loading: boolean
+  error: Error | null
 }
 
-export const useNetworkName = (lotusApiAddr: string) => {
-  const {
-    data: networkName = '',
-    error = null,
-    isValidating
-  } = useSWR(lotusApiAddr, fetcher)
+const fetcher = async (
+  lotusApi: LotusRpcEngine,
+  lotusMethod: string
+): Promise<string> => await lotusApi.request<string>(lotusMethod)
+
+export const useNetworkName = (): UseNetworkNameResult => {
+  const { lotusApi } = useEnvironment()
+  const { data, error } = useSWR<string, Error>(
+    [lotusApi, 'StateNetworkName'],
```

Comment:
I'm not sure about passing an object as the key here... without reading the docs, gut instincts tell me to use the `lotusApiAddr` string here and instantiate the lotusApi inside the `fetcher`

I know they use those keys for caching

Reply:
Yeah I think you're right it's better to use the string. In theory it's fine to use the lotusApi as a key, but since the `lotusApi` is reinstantiated between network switches, it won't keep the cache. When using the api URL as string, the cache would still be valid after switching back to a network.